Day1/P1.py

```diff
@@ -0,0 +1,154 @@
+print(findchar('hello my name is tram','m',8,12))
+print('-'*20)
+
+#find the positions of all occurrences of a character in a string
+def findchars(txt,char,start=None,end=None):
+    myindex = []
+    if start is None:
+        start = 0
+    if end is None:
+        end = len(txt)
+    for i in range(start,end):
+        if txt[i:i+len(char)] == char:
+            myindex.append(i)
+    return myindex
+
+print(findchars('hello my name is tram','a'))
+print(findchars('hello my name is tram','my'))
+print(findchars('hello my name is tram','z'))
+print(findchars('hello my name is tram','lo'))
+print(findchars('hello my name is tram','a',12))
+print(findchars('hello my name is tram','m',8,12))
+print(findchars('ohmymymyabcdefmy','my'))
+print('-'*20)
+
+#Convert string into lower case
+txt = 'Hello, And Welcome To My World!'
+print(txt.lower())
+
+def lowerstr(txt):
+    uppertolower = {'A':'a','B':'b','C':'c','D':'d','E':'e','F':'f','G':'g','H':'h','I':'i','J':'j','K':'k','L':'l','M':'m','N':'n','O':'o','P':'p','Q':'q','R':'r','S':'s','T':'t','U':'u','V':'v','W':'w','X':'x','Y':'y','Z':'z'}
+    for i in range(len(txt)):
+        if txt[i] in uppertolower.keys():
+            txt = txt[0:i] + uppertolower[txt[i]] + txt[i+1:]
+    return txt
+
+print(lowerstr('Hello, And Welcome To My World!'))
+print(lowerstr('HSKHNKXjkjskfeKWEOIIHF'))
+print(lowerstr('hello how are you'))
+print('-'*20)
+
+#upper case all characters
+def upperstr(txt):
+    lowertoupper = {'a':'A','b':'B','c':'C','d':'D','e':'E','f':'F','g':'G','h':'H','i':'I','j':'J','k':'K','l':'L','m':'M','n':'N','o':'O','p':'P','q':'Q','r':'R','s':'S','t':'T','u':'U','v':'V','w':'W','x':'X','y':'Y','z':'Z'}
+    for i in range(len(txt)):
+        if txt[i] in lowertoupper.keys():
+            txt = txt[0:i] + lowertoupper[txt[i]] + txt[i+1:]
+    return txt
+
+print(upperstr('Hello, And Welcome To My World!'))
+print(upperstr('HSKHNKXjkjskfeKWEOIIHF'))
+print(upperstr('HELLO HOW ARE YOU'))
+print('-'*20)
+
+#count the number of times a specified value appears in the string
+def countvalue(txt,char,start=None,end=None):
+    count = 0
+    for i in findchars(txt,char,start,end):
+        count += 1
+    return count
+
+print(countvalue('hello my name is tram','a'))
+print(countvalue('hello my name is tram','my'))
+print(countvalue('hello my name is tram','z'))
+print(countvalue('hello my name is tram','lo'))
+print(countvalue('hello my name is tram','a',12))
+print(countvalue('hello my name is tram','m',8,12))
+print(countvalue('ohmymymyabcdefmy','my'))
+print(countvalue('I love apples, apple are my favorite fruit','apple'))
+print('-'*20)
+
+#string capitalize method: change first character in a string to upper case, while other characters remain lower case
+def capitalizestr(txt):
+    alphabet = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
+    if txt[0] in alphabet:
+        txt = upperstr(txt[0]) + txt[1:]
```

Comment:
If you are gonna use `upperstr` to help you here, why do you even need to bother with `alphabet`? `upperstr` function should be able to change all valid characters, and handle the non-valid case for you